lib/fog/sakuracloud/models/volume/disk.rb

```diff
@@ -20,8 +20,12 @@ def delete
         alias_method :destroy, :delete
 
         def save
-          requires :name, :plan, :source_archive
-          data = service.create_disk(@attributes[:name], @attributes[:plan], @attributes[:source_archive]).body["Disk"]
+          requires :name, :plan
+          options = {
+            :source_archive=>@attributes[:source_archive],
+            :size_mb=>@attributes[:size_mb]
+          } 
+          data = service.create_disk(@attributes[:name], @attributes[:plan], options).body["Disk"]
```

Comment:
Also i would say that you should not work towards parsing of the request in your model. The request should do it for you. So `.body["Disk"]` should be the return of the method `create_disk`.